File: saude_java/app/src/test/java/saude_java/TMBTest.java
package saude_java;

import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

public class TMBTest {

    @Test
    public void testCalcularTMBMasculino() {
        assertEquals(1695.36, TMB.calcular(70, 175, 30, "masculino"));
    }

    @Test
    public void testCalcularTMBFeminino() {
        assertEquals(1505.1, TMB.calcular(70, 175, 30, "feminino"));
    }

    @Test
    public void testSexoInvalido() {
        Exception exception = assertThrows(IllegalArgumentException.class, () -> {
            TMB.calcular(70, 175, 30, "outro");
        });
        assertEquals("Sexo deve ser 'masculino' ou 'feminino'", exception.getMessage());
    }
}

File: saude_php/tests/TMBTest.php
<?php

use PHPUnit\Framework\TestCase;
use SaudePhp\TMB;

class TMBTest extends TestCase
{
    public function testCalcularTMBMasculino()
    {
        $tmb = TMB::calcular(70, 175, 30, 'masculino');
        $this->assertEquals(1695.36, $tmb);
    }

    public function testCalcularTMBFeminino()
    {
        $tmb = TMB::calcular(60, 160, 25, 'feminino');
        $this->assertEquals(1388.1, $tmb);
    }

    public function testSexoInvalido()
    {
        $this->expectException(\InvalidArgumentException::class);
        TMB::calcular(70, 175, 30, 'invalido');
    }
}
